### PR TITLE
[jvm] generate interface-per-typedef

### PR DESCRIPTION
**Pull request by Simn**
I'm too tired to write a lot of stuff right now but this brings the class-to-anon call benchmarks from 5M samples to 120M samples. Probably doesn't help much with real code though.

### CODE REVIEW

File: src/generators/genjvm.ml [hunk 1]
```diff
@@ -73,6 +73,7 @@ type generation_context = {
 	anon_identification : jsignature tanon_identification;
 	preprocessor : jsignature preprocessor;
 	default_export_config : export_config;
+	nadako : jsignature nadako_interfaces;
```
Comment (Aurel300) [inline src/generators/genjvm.ml]: I don't want to not give credit to nadako, but let's have a proper name for this …

File: src/generators/genjvm.ml [hunk 2]
```diff
@@ -70,10 +70,10 @@ type generation_context = {
 	jar : Zip.out_file;
 	t_exception : Type.t;
 	t_throwable : Type.t;
-	anon_identification : jsignature tanon_identification;
-	preprocessor : jsignature preprocessor;
+	mutable anon_identification : jsignature tanon_identification;
+	mutable preprocessor : jsignature preprocessor;
 	default_export_config : export_config;
-	nadako : jsignature nadako_interfaces;
+	mutable nadako : jsignature nadako_interfaces;
```
Comment (nadako) [inline src/generators/genjvm.ml]: > mutable nadako

that makes me uncomfortable ^^

Comment (Gama11) [reply src/generators/genjvm.ml]: Would you rather be immutable?

Comment (nadako) [reply src/generators/genjvm.ml]: I prefer not to mutate for now

Comment (kevinresol) [reply src/generators/genjvm.ml]: does that mean cancer

File: src/generators/genjvm.ml (hunk 1 above)
Comment (kevinresol) [reply src/generators/genjvm.ml]: this is how asteroids are named after the discoverer

**Pull request merged**